compiler/dalgona/src/HDF5Importer.h

```diff
@@ -0,0 +1,82 @@
+/*
+ * Copyright (c) 2020 Samsung Electronics Co., Ltd. All Rights Reserved
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef __DALGONA_HDF5IMPORTER_H__
+#define __DALGONA_HDF5IMPORTER_H__
+
+#include <luci_interpreter/core/Tensor.h>
+
+#include <H5Cpp.h>
+
+using Shape = luci_interpreter::Shape;
+using DataType = luci_interpreter::DataType;
+
+namespace dalgona
+{
+
+// HDF5Importer reads an input data saved in the hdf5 file in the given path
+// The hierarchy of the hdf5 file is as follows.
+// Group "/"
+//  > Group "value"
+//    > Group <record_idx>
+//      > Dataset <input_idx>
+// record_idx : index of the record (dataset file can contain multiple records)
+// input_idx : index of the input (DNN model can have multiple inputs)
+// Ex: the j'th input of the i'th record can be accessed by "/value/i/j"
+class HDF5Importer
+{
+public:
+  explicit HDF5Importer(const std::string &path) : _file{path, H5F_ACC_RDONLY}
+  {
+    // Do nothing
+  }
+
+public:
+  /**
+   * @brief importGroup has to be called before readTensor is called
+   *        Otherwise, readTensor will throw an exception
+   */
+  void importGroup() { _value_grp = _file.openGroup("value"); }
```

Comment:
(optional) I'm not sure(need to test) but data or group can be accessed by its absolute path as well. Maybe absolute path makes this line be removed and prevent from mistakes of developers. What I think is like
```cxx
void HDF5Importer::readTensor(int32_t record_idx, int32_t input_idx, void *buffer)
{
  // auto record = _value_grp.openGroup(std::to_string(record_idx));
  auto tensor = record.openDataSet("value" + std::to_string(record_idx) + std::to_string(input_idx));

  readTensorData(tensor, static_cast<uint8_t *>(buffer));
}
```

Reply:
It looks much better. I've changed readTensor. Thanks.

Reply:
Maybe this method can be removed now?

Reply:
I removed `importGroup` and `_value_grp` member variable. Thanks!